lang/cs/Org.Apache.REEF.IO/FileSystem/AzureBlob/AzureBlobFileSystem.cs

```diff
@@ -0,0 +1,255 @@
+            {
+                container.DeleteIfExists();
+                return;
+            }
+
+            var directory = container.GetDirectoryReference(uriSplit[1]);
+
+            for (var i = 2; i < uriSplit.Length; i++)
+            {
+                directory = directory.GetDirectoryReference(uriSplit[i]);
+            }
+
+            foreach (var blob in directory.ListBlobs(true).OfType<ICloudBlob>())
+            {
+                blob.DeleteIfExists();
+            }
+        }
+
+        /// <summary>
+        /// Gets the children of the blob "directory."
+        /// </summary>
+        public IEnumerable<Uri> GetChildren(Uri directoryUri)
+        {
+            BlobContinuationToken blobContinuationToken = null;
+            var path = directoryUri.AbsolutePath.Trim('/');
+
+            do
+            {
+                var listing = _client.ListBlobsSegmented(path, false,
+                    BlobListingDetails.None, null,
+                    blobContinuationToken, new BlobRequestOptions(), new OperationContext());
+
+                if (listing.Results != null)
+                {
+                    foreach (var listBlobItem in listing.Results)
+                    {
+                        yield return listBlobItem.Uri;
+                    }
+                }
+
+                blobContinuationToken = listing.ContinuationToken;
+            } while (blobContinuationToken != null);
+        }
+
+        /// <summary>
+        /// Creates a Uri using the relative path to the remote file (including the container),
+        /// getting the absolute URI from the Blob client's base URI.
+        /// </summary>
+        /// <param name="path">The relative path to the remote file, including the container</param>
+        /// <returns>The URI to the remote file</returns>
+        public Uri CreateUriForPath(string path)
+        {
+            return new Uri(_client.BaseUri.AbsoluteUri.TrimEnd('/') + '/' + path.Trim('/'));
+        }
+
+        /// <summary>
+        /// Gets the FileStatus based on reports from Blob. DateTime of the FileStatus
+        /// is based on UTC.
+        /// </summary>
+        public FileStatus GetFileStatus(Uri remoteFileUri)
+        {
+            var blobReference = _client.GetBlobReferenceFromServer(remoteFileUri);
+            if (!blobReference.Exists())
+            {
+                throw new StorageException("Unable to find blob at " + remoteFileUri);
+            }
+
+            blobReference.FetchAttributes();
+
+            var lastModifiedTime = blobReference.Properties.LastModified;
+            if (!lastModifiedTime.HasValue)
+            {
+                throw new StorageException("Blob at " + remoteFileUri + " does not have a last modified" +
+                                           "time. It may have been deleted.");
+            }
+
+            return new FileStatus(lastModifiedTime.Value.DateTime.ToUniversalTime(), blobReference.Properties.Length);
```

Comment:
You're right, if YARN submission requires universal time, perhaps we should do it at another layer instead.